Mathlib/Analysis/LegendrePoly.lean

```diff
@@ -0,0 +1,138 @@
+/-
+Copyright (c) 2024 Junqi Liu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Junqi Liu
+-/
+import Mathlib.Algebra.Polynomial.Basic
+import Mathlib.Algebra.BigOperators.Ring
+import Mathlib.Algebra.Order.BigOperators.Group.Finset
+import Mathlib.Algebra.Polynomial.AlgebraMap
+import Mathlib.Algebra.Polynomial.Derivative
+import Mathlib.Analysis.Normed.Module.Basic
+import Mathlib.Data.Nat.Choose.Sum
+import Mathlib.Data.Real.Basic
+import Mathlib.Tactic.Linarith
+import Mathlib.Tactic.Ring
+
+/-!
+# Legendre Polynomials
+
+In this file, we define the Legendre polynomials `Legendre n` for `n : ℕ` as a polynomial in `ℝ[X]`.
+We prove some basic properties of the Legendre polynomials.
+
+-/
+
+open scoped Nat
+open BigOperators Finset
+
+variable {R : Type*}
+
+namespace Polynomial
+
+/--  `Legendre n` is the polynomial defined in terms of derivatives of order n.  -/
+noncomputable def Legendre (n : ℕ) : ℝ[X] :=
+  C (n ! : ℝ)⁻¹ * derivative^[n] (X ^ n * (1 - X) ^ n)
+
+lemma Finsum_iterate_deriv [CommRing R] (k : ℕ) (h : ℕ → ℕ) :
+    derivative^[k] (∑ m in Finset.range (k + 1), (h m) • ((- 1) ^ m : R[X]) * X ^ (k + m)) =
+    ∑ m in Finset.range (k + 1), (h m) • (- 1) ^ m * derivative^[k] (X ^ (k + m)) := by
+  induction' k + 1 with n hn
+  · simp only [Nat.zero_eq, Finset.range_zero, Finset.sum_empty, iterate_map_zero]
+  · rw[Finset.sum_range, Finset.sum_range, Fin.sum_univ_castSucc, Fin.sum_univ_castSucc] at *
+    simp only [Fin.coe_castSucc, Fin.val_last, iterate_map_add, hn, add_right_inj]
+    rw [nsmul_eq_mul, mul_assoc, ← nsmul_eq_mul, Polynomial.iterate_derivative_smul, nsmul_eq_mul,
+      mul_assoc]
+    rcases n.even_or_odd with (hn1 | hn2)
+    · simp_all only [nsmul_eq_mul, Int.even_coe_nat, Even.neg_pow, one_pow, one_mul]
+    · rw [Odd.neg_one_pow]
+      simp only [neg_mul, one_mul, iterate_map_neg, mul_neg]
+      exact_mod_cast hn2
+
+/-- The expand of `Legendre n`. -/
+theorem Legendre_eq_sum (n : ℕ) : Legendre n = ∑ k in Finset.range (n + 1),
+    C ((- 1) ^ k : ℝ) • (Nat.choose n k) * (Nat.choose (n + k) n) * X ^ k := by
```

Comment:
I am a bit confused. This suggests that the coefficients of `Legendre` are integers. According to https://en.wikipedia.org/wiki/Legendre_polynomials#Rodrigues'_formula_and_other_explicit_formulas some coefficients are fractions with a power of two in the denominator.

Are these different definitions of the Legendre polynomials?

Reply:
emmmm. oh! By this wikipedia, the polynomial which is defined may be called shifted Legendre Polynomial. Maybe the name which I used was wrong.

Reply:
I see. So let's rename these to `shiftedLegendre`. And could you then please make them polynomials that by definition take values in `\Z` instead of `\R`?

Reply:
ok! I have fixed by your suggestion. I'll push it after I push iterate derivative of finset sum.